rir/src/compiler/analysis/verifier.cpp

```diff
@@ -163,7 +162,8 @@ class TheVerifier {
 
         if (auto mk = MkArg::Cast(i)) {
             auto p = mk->prom();
-            assert(p->owner->promise(p->id) == p);
+            if (p->owner->promise(p->id) != p)
```

Comment:
sorry, I don't want to drag this out unnecessarily. But if you actually look at what the verifier does, whenever there is a verification error, it sets ok to false. Then the callees currently do `assert(Verfier::verify....`.

So although you now changed a bunch of assumes into errors, you do not check the final verifier result with those changes.

can you please (instead of erroring on some conditions inside the verification functions), just error on the final `ok` result??